kolibri/plugins/media_player/assets/src/modules/captions/index.js

```diff
@@ -188,14 +177,31 @@ export default {
       store.dispatch('synchronizeTrackList');
     },
     setCuesFromTrack(store, track) {
-      store.commit('SET_CUE_LIST', track.cues);
+      store.commit('DELETE_CUES');
+
+      const cues = trackUtils.listToArray(track.cues || []);
       // Ensure cues have ids
-      store.getters.cues.forEach((cue, i) => {
+      cues.forEach((cue, i) => {
         cue.id = track.id + '-cue-' + i;
       });
+
+      store.commit('SET_CUES', cues);
     },
     setActiveCuesFromTrack(store, track) {
-      store.commit('SET_ACTIVE_CUE_LIST', track.activeCues);
+      // In case we get triggered to set active cues but haven't added cues yet, do that now
+      // This helps an issue in Safari where we don't get an updated cue list
+      if (track.cues && track.cues.length !== store.state.cues.length) {
+        store.dispatch('setCuesFromTrack', track);
+      }
```

Comment:
cool

Reply:
It doesn't entirely fix the issue in Safari, but it helps. I'll definitely add a followup issue regarding it.